presto-docs/src/main/sphinx/develop/functions.rst

```diff
@@ -2,31 +2,68 @@
 Functions
 =========
 
+Plugin Implementation
+---------------------
 The function framework is used to implement SQL functions. Presto includes a
-number of built-in functions, and an internal ``Plugin`` (plugins that have a
-dependency on presto-main) can provide new functions by returning a
-``FunctionFactory`` from ``getServices()``.
+number of built-in functions. In order to implement new functions, you can
+write a plugin that returns one more more functions from ``getFunctions()``. For example:
 
 .. code-block:: java
 
-    @ScalarFunction("is_null")
-    @Description("Returns TRUE if the argument is NULL")
-    @SqlType(StandardTypes.Boolean)
-    public static boolean isNull(@SqlNullable @SqlType(StandardTypes.VARCHAR) Slice string)
+    public class ExampleFunctionsPlugin
+        implements Plugin
+    {
+        @Override
+        public Set<Class<?>> getFunctions()
+        {
+            return ImmutableSet.<Class<?>>builder()
+                    .add(ExampleNullFunction.class)
+                    .add(IsNullFunction.class)
+                    .add(IsEqualOrNullFunction.class)
+                    .add(ExampleStringFunction.class)
+                    .add(ExampleAverageFunction.class)
+                    .build();
+        }
+    }
+
+
+Note that the ``ImmutableSet`` class is a utility class from Guava.
+``getFunctions()`` contains all of the functions that we will implement below
+in this tutorial.
+
+For a full example in the codebase, see either the ``presto-ml`` directory for machine
+learning functions or the ``presto-teradata-functions`` directory for Teradata-compatible
+functions, both in the root of the Presto source.
+
+Scalar Function Implementation
+------------------------------
+The function framework uses annotations to indicate relevant information
+about functions, including name, description, return type, and parameter
```

Comment:
We don't use a comma before "and" in the documentation (unless required due to ambiguity).